Art2om/JPG-to-ASCII-Artwork-Converter
# Add ASCII Artwork Converter Class and CLI Support

This PR introduces a complete ASCII Artwork Converter implementation, alongside CLI support for seamless usage. The changes include foundational setup, ASCII conversion logic, and documentation updates. Key highlights are as follows:

## Summary of Changes
- Initial Repository Setup
  - Initialized a Python project setup with configuration files for dependencies and repository structure.
  - Added core dependencies required for ASCII conversion and CLI functionality.
- Core Features and Implementations
  - ASCII Converter Class: Implemented JPGToASCIIConverter, a robust class for converting .jpg images to ASCII format.
  - CLI Scripts: Added CLI command jpg_ascii, which accepts image_path, scale_factor, single_char_height, and single_char_width parameters for customized ASCII output.
  - Constants: Defined core constants for customizable aspects of ASCII art (e.g., character height and width, scale factor, font, and character set).
- Documentation and Testing
  - Enhanced documentation to provide setup instructions, usage examples, and explanations of configurable parameters.
  - Implemented initial tests for the ASCII conversion class to verify functionality and stability.

## Commits
a75ac1a: refactor - Removed base proof of concept code.
4d528e5: chore(docs) - Improved documentation for project setup and CLI usage.
676d390: feat - Added CLI scripts for the jpg_ascii command.
824c4f5: test - Created tests for the JPGToASCIIConverter class.
39d261c: feat - Added JPGToASCIIConverter class to handle conversion logic.
5143dd5: feat - Defined constants for ASCII conversion.
43141e1: chore(deps) - Added initial dependencies.
9656f5a: chore - Initialized Python setup configuration.
9eebcfe: chore - Prepared repository for Python development.

## Testing Instructions:

1. Install dependencies as specified.
2. Use jpg_ascii <image_path> <scale_factor> <single_char_height> <single_char_width> to test ASCII conversion from the command line.
3. Run included tests to verify functionality.

This PR refines the project structure, introduces essential functionality for ASCII art conversion, and enhances usability with a comprehensive CLI tool.
